0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,190 @@
+---
+title: SPSP Pull Payments
+draft: 1
+---
+# SPSP Pull Payments
+
+## Preface
+
+This document describes how to conduct pull payments via [STREAM](../0029-stream/0029-stream.md) once the payment details have been exchanged via the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md) (SPSP).
+
+## Introduction
+
+### Motivation
+
+The Simple Payment Setup Protocol (SPSP) only descirbes how the details required for a STREAM connection are exchanged. However, it lacks a detailed explaination of how a pull payment is conducted using this connection. 
+
+### Scope
+
+This document specifies basic server details needed by the client to set up a STREAM pull payment. It is intended for use by end-user applications.
+
+### Operation
+
+Any SPSP server will expose an HTTPS endpoint called the SPSP Endpoint. The client can query this endpoint to get information about about the server's connection details, namely ILP address and a shared secret, as well as additional information about the type and form of payment that can be made to this server. The client can set up and pull multiple ILP payments using the details provided by the server. 
+
+## Specification
+
+### Query (`GET <SPSP Endpoint>`)
+
+The client queries the SPSP endpoint to get information about the server:
+
+#### Request
+
+``` http
+GET /pullagreement456 HTTP/1.1
+Host: alice.com
+Accept: application/spsp4+json, application/spsp+json
+```
+
+#### Response
+
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/spsp4+json
+
+{
+  "destination_account": "alice.ilpdemo.red~pullagreement456",
+  "shared_secret": "k5nubgM6zpb88NPGVnI/tVjRdgpUh+JvMueRFEMvPcY=",
+  "balance": {
+    "current": "100",
+    "maximum": "500"
+  },
+  "asset_info": {
+    "code": "USD",
+    "scale": 2
+  },
+  "frequency_info": { 
+    "type": "MONTH",
+    "interval": 1
+  },
+  "timeline_info": {
+    "refill_time": "2019-02-10T01:01:13Z",
+    "expiry_time": "2019-07-10T01:01:12Z"
+  },
+  "receiver_info": {
```

Comment:
This is a new specification. We don't need to use `receiver_info` for backwards compatibility and this is actually the sender.

Can we use a more neutral term like "counterparty"?

Reply:
I actually had the receiver in mind. This could be useful for an end user application (a wallet supporting pull payments) to display details about this particular pull agreement. But maybe there should be a field called `counterparty_info` that contains the details about sender and receiver.